code/game/machinery/computer/buildandrepair.dm

```diff
@@ -22,12 +109,49 @@
 	if(state != FRAME_COMPUTER_STATE_EMPTY)
 		balloon_alert(user, "circuit already installed!")
 		return FALSE
-	return ..()
+	if(!anchored && istype(board))
+		balloon_alert(user, "frame must be anchored!")
+		return FALSE
+	. = ..()
+	if(. && !by_hand)
+		state = FRAME_COMPUTER_STATE_BOARD_SECURED
+		update_appearance(UPDATE_ICON_STATE)
+
+/obj/structure/frame/computer/install_parts_from_part_replacer(mob/living/user, obj/item/storage/part_replacer/replacer, no_sound = FALSE)
+	switch(state)
+		if(FRAME_COMPUTER_STATE_BOARD_SECURED)
+			var/obj/item/stack/cable_coil/cable = locate() in replacer
+			if(isnull(cable))
+				return FALSE
+
+			if(add_cabling(user, cable, time = 0))
+				if(!no_sound)
+					replacer.play_rped_sound()
+					if(replacer.works_from_distance)
+						user.Beam(src, icon_state = "rped_upgrade", time = 0.5 SECONDS)
+					no_sound = TRUE
+				return install_parts_from_part_replacer(user, replacer, no_sound = no_sound)  // Recursive call to handle the next part
+
+			return FALSE
+
+		if(FRAME_COMPUTER_STATE_WIRED)
+			var/obj/item/stack/sheet/glass/glass_sheets = locate() in replacer
+			if(isnull(glass_sheets))
+				return FALSE
+
+			if(add_glass(user, glass_sheets, time = 0))
+				if(!no_sound)
+					replacer.play_rped_sound()
+					if(replacer.works_from_distance)
+						user.Beam(src, icon_state = "rped_upgrade", time = 0.5 SECONDS)
+				return TRUE
+
+			return FALSE
 
 /obj/structure/frame/computer/item_interaction(mob/living/user, obj/item/tool, list/modifiers, is_right_clicking)
 	. = ..()
 	if(. & ITEM_INTERACT_ANY_BLOCKER)
-		return .
+		return
```

Comment:
We keep this as per style guide https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#dot-variable-
```suggestion
		return .
```